ocaml/idl/datamodel_errors.ml

```diff
@@ -1708,8 +1708,8 @@ let _ =
     ~doc:"The provided certificate has expired." () ;
   error Api_errors.server_certificate_signature_not_supported []
     ~doc:
-      "The provided certificate is not using the SHA256 (SHA2) signature \
-       algorithm."
+      "The provided certificate is not using one of the following signature \
+       algorithms:  SHA256 (SHA2), SHA512."
```

Comment:
SHA-512 and SHA-256 are both part of the SHA-2 family. There are other members of the SHA-2 family that we don't support though (SHA-384, and also some less secure variants SHA-224, SHA-512/256, etc.).

perhaps move the `(SHA2)` earlier `one of the following SHA-2 signature algorithms`

Reply:
SHA naming is horrible.  https://en.wikipedia.org/wiki/SHA-2#Comparison_of_SHA_functions is the best table I've found.

SHA1 is a single algorithm.  SHA2 comes in multiple digest sizes.  SHA3 is different again, and comes in multiple digest sizes which also alias the SHA2 digest sizes.

It is critical to disambiguate SHA2-256/512 from SHA3-256/512 because they're very different things.